### PR TITLE
Adds document and script describing the creation of OSC releases

### PR DESCRIPTION
**Pull request by emanoelxavier**
1. Script for creation of release branches, release tags and snapshot.
2. Description of the OSC release concepts and documented steps to create a release.
3. Updates other md links.

### CODE REVIEW

File: releases/scripts/create-release.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        self.name = name  # name of the repo
+        self.version = version # version of the repo to be used in the pom.xml, branch name, git tag, ...
+        self.sec_mgr_version = sec_mgr_version # version of the security mgr SDK to be used in the dependencies
+        self.sdn_ctlr_version = sdn_ctrlr_version # version of the SDN ctrlr  SDK to be used in the dependencies
+
+    # Creates a new release branch with the updated and committed pom
+    # according to the version information in self
+    def prepare_release_branch(self):
+        print("Creating the release branch for {}\n".format(self))
+        sys.stdout.flush()
```
Comment (arvindn05) [inline releases/scripts/create-release.py]: use below to always flush
```
import functools
print = functools.partial(print, flush=True)
```
or use
`print('foo', flush=False)`

File: releases/scripts/create-release.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+        # Creating the release branch
+        self.process_run(p_args=["git", "checkout", "-b", self.version])
+
+        # Creating the version commit with modified pom
+        self.create_version_commit()
+
+        # Upstreaming the release branch
+        self.process_run(p_args=["git", "push", "origin", self.version, "-f"]) #upstream
```
Comment (arvindn05) [inline releases/scripts/create-release.py]: do we need force push?

File: releases/scripts/create-release.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            Repo("osc-nuage-plugin", sdn_ctrlr_api_release_version, "n/a", sdn_ctrlr_api_release_version),
+        ]
+
+
+def add_header(filename):
+    with open(filename, 'r+') as f:
+        content = f.read()
+        f.seek(0,0)
+        f.write("""<!--
+    Copyright (c) Intel Corporation
```
[the remark this answers is not in the document]
Comment (emanoelxavier) [reply releases/scripts/create-release.py]: Agreed, this is only called if not nuage or pan. see
 ```py
        if self.name not in ("osc-pan-plugin", "osc-nuage-plugin"):
            add_header(pom_file)
```

File: releases/scripts/create-release.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+    def process_Popen(self, p_args, p_stdout):
+        return subprocess.Popen(p_args, cwd=self.name, stdout=p_stdout)
+
+    def __str__(self):
+        return "Repo {} with version {}, sec-mgr- version {}, sdn-ctrlr-version {}\n".format(self.name, self.version, self.sec_mgr_version, self.sdn_ctlr_version)
+
+
+# Desired snapshot versions for each repo
+repo_snapshots = [
```
[the remark this answers is not in the document]
Comment (emanoelxavier) [reply releases/scripts/create-release.py]: Looks like these have to come after the class Repo is defined or else we get 'Repo not defined`

**Pull request merged**